src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/SigningInterceptor.kt

```diff
@@ -0,0 +1,23 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import okhttp3.HttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+
+class SigningInterceptor(private val parameterName: String, private val signer: HttpUrl.() -> String) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val userRequest: Request = chain.request()
```

Comment:
Would it be a bit more simple to use `with(chain.request) { }`?
And the same thoughts for builders

Reply:
Agreed, don't know how I missed that one! 😅